### ISSUE TITLE
Group numbers for the lanthanides and actinides are strange

### ISSUE THREAD

**Issue by arosen93** [materialsproject/pymatgen#2327]
When accessing the `.group` property of a `specie` that is a lanthanide or actinide, the returned group number is quite unconventional. For instance, the group number for Yb is 16. For Lu it's 17. While there is certainly debate to be had around the group numbers for the lanthanides and actinides, I don't believe this choice would ever be considered one of the [conventional options](https://www.chemeurope.com/en/encyclopedia/Group_number_of_lanthanides_and_actinides.html). 

```python
from pymatgen.core import Structure
struct = Structure(lattice=[[0, 2.13, 2.13], [2.13, 0, 2.13], [2.13, 2.13, 0]],species=["Yb", "O"],coords=[[0, 0, 0], [0.5, 0.5, 0.5]])
print(struct[0].specie.group)
# ^outputs 16
```

**Comment by shyuep** [comment]
Thanks. Feel free to propose a conventional and implement a change. For me, group numbers don't really make any kind of sense of the TM and rare earth series. If you are planning to use it for ML purposes, I would say group number is probably one of the worst choices you can use. I for one don't believe the large body of work out there that utilizes this kind of features for ML.

**Comment by arosen93** [comment]
I definitely agree with you re: group number being used as a descriptor for ML. Besides, it's often highly correlated with many other properties anyway.

In this case, I wasn't even thinking of ML. If one wanted to query a database of structures to find all the entries containing halogens, one might write a query that looks for group 17 elements. But Lu and Lr would unexpectedly come up too. There are of course better ways to do the query, but we should make it easy on the user.

To back up what I'm going to suggest, here are a few points:
- IUPAC understandably makes no explicit statement on the group number.
- Currently, in pymatgen La and Ac are group 3 whereas Lu and Lr are group 17. However, Lu and Lr are more group-3-like than La and Ac, as argued [here](https://pubs.acs.org/doi/10.1021/ed059p634), so this is a point that should be resolved in a proposed change.
- The lanthanoids and actinoids share common chemistry and should, in my view, not have different group numbers. 

With the above points, the only solutions I can think of are:
1. All lanthanoids and actinoids are in group 3.
2. All lanthanoids and actinoids are in a ficticious group 19, which would be mentioned in the documentation.
3. All lanthanoids and actinoids are not in group 3 but have `None` as the group number. This will likely pose a problem for down-stream ML tasks, and I do not advise it.
4. Lu and Lr are in group 3, but the remaining lanthanoids and actinoids have `None` as the group number. Again, same problem as Option 3.

@shyuep: **Do you have a strong opinion on Option 1 vs. Option 2**?

**Comment by shyuep** [comment]
I don't have a strong opinion, but I would note that the group is used to generate the periodic table layout. I don't mind them having an effective group number of 3. In any case, I wouldn't have used group 17 for halogens but rather specify the exact set of element (F, Cl, Br, I).

**Comment by arosen93** [comment]
Closed in #2343.
